misoc/software/include/base/irq.h

```diff
@@ -44,6 +81,8 @@ static inline unsigned int irq_getmask(void)
 	return mask;
 #elif defined (__or1k__)
 	return mfspr(SPR_PICMR);
+#elif defined (__vexriscv__)
+    return read_csr(0x330);
```

Comment:
We shouldn't have magic values, add a header with register definitions instead.

Reply:
system.h too ?

Reply:
Sure

Reply:
Hm no, not system.h, there is `spr-defs.h` for or1k, so `csr-defs.h` for riscv should make sense.

Reply:
Unfortunately : 
read_csr(DEFINED_CONSTANT) 
-> Error: Instruction csrr requires absolute expression

Also, can't do as the or1k do, as the read_csr macro require compilation time constants.

Reply:
You can just leave out the __builtin_constant_p branch. The miniscule reduction in performance doesn't matter.

Reply:
Unfortunatly, we can't do the same way than or1k, as RISC-V do not provide a instruction to access CSR pointed by a CPU register.

The CSR address realy need to be known at compilation time.
=> asm volatile ("csrr %0, %1" : "=r"(mask) : "i"(CSR_IRQ_MASK));   + csr-defs.h

Reply:
oh WTF gross. Sure, keep them as macros, then, I guess, no other choice...

@jordens One more reason to not have mtspr/mfspr in ARTIQ Python.